### ISSUE TITLE
Share snapshot ID across AccountDB and StorageDB

### ISSUE THREAD

**Issue by carver** [ethereum/py-evm#1760]
> The same trick of sharing snapshot ids might be possible [in the linked comment].

_Originally posted by @lithp in https://github.com/ethereum/py-evm/pull/1735_

Then the snapshots go back to being a single state root and UUID. Seems like a good thing.

**Comment by carver** [comment]
After #1735 is merged, this would be a reasonable first introduction to the repo.

**Comment by carver** [comment]
Oops, I think I closed it in #1735 -- anyway, it's already done.
